server/middlewares/rateLimiter.ts

```diff
@@ -35,7 +41,20 @@ export function rateLimiter() {
   };
 }
 
-export function registerRateLimiter(config: RateLimiterConfig) {
+type RateLimiterConfig = {
+  /** The window for which this rate limiter is considered (defaults to 60s) */
+  duration?: number;
+  /** The number of requests per IP address that are allowed within the window */
+  requests?: number;
```

Comment:
How about marking it `required`? Usage of `rateLimiter({})` should be prevented if there's no need to enforce readability.

Reply:
Ah yes will do